LemonUI/Scaleform/Celebration.cs

```diff
@@ -0,0 +1,502 @@
+            {
+                _celeb.Visible = value;
+                _celebBg.Visible = value;
+                _celebFg.Visible = value;
+            }
+        }
+
+        /// <summary>
+        /// Creates a standard Celebration with a duration of 3
+        /// </summary>
+        /// <footer><a href="https://www.google.com/search?q=LemonUI.Scaleform.Celebration">`Celebration` on google.com</a></footer>
+        public Celebration() : this(3)
+        {
+        }
+
+        /// <summary>
+        /// Creates a custom Celebration with the specified duration.
+        /// </summary>
+        /// <param name="duration">The duration of each item on the wall.</param>
+        /// <footer><a href="https://www.google.com/search?q=LemonUI.Scaleform.Celebration">`Celebration` on google.com</a></footer>
+        public Celebration(int duration)
+        {
+            _items = new List<Action<CelebrationPart>>();
+
+            _celeb = new CelebrationPart(CelebrationLayer.Main, _items);
+            _celebBg = new CelebrationPart(CelebrationLayer.Background, _items);
+            _celebFg = new CelebrationPart(CelebrationLayer.Foreground, _items);
+
+            Duration = duration;
+        }
+
+        /// <summary>
+        /// Processes the object.
+        /// </summary>
+        /// <footer><a href="https://www.google.com/search?q=LemonUI.Scaleform.Celebration.Process">`Celebration.Process` on google.com</a></footer>
+        public void Process()
+        {
+            if (!Visible) return;
+
+            _celebBg?.Process();
+            _celeb?.Process();
+            _celebFg?.Process();
+        }
+
+        /// <summary>
+        /// Remove all the items on the wall.
+        /// </summary>
+        /// <footer><a href="https://www.google.com/search?q=LemonUI.Scaleform.Celebration.Reset">`Celebration.Reset` on google.com</a></footer>
+        public void Reset()
+        {
+            _items.Clear();
+        }
+
+        /// <summary>
+        /// Add a Winner item to the wall.
+        /// </summary>
+        /// <param name="hasWon">Whether the player has won.</param>
+        /// <param name="gamerName">The element that has been won (Game, Round, Match, etc).</param>
+        /// <param name="teamName">The team that has won the game.</param>
+        /// <param name="crewName">The crew that has won the game.</param>
+        /// <param name="betAmount">The amount that the player has gained/lost in betting (0 = off).</param>
+        /// <param name="rawGamerName">Whether the gamerName is rawText (null = automatic).</param>
+        /// <footer><a href="https://www.google.com/search?q=LemonUI.Scaleform.Celebration.AddWinner">`Celebration.AddWinner` on google.com</a></footer>
+        public void AddWinner(bool hasWon, string gamerName = "CELEB_ROUND", string teamName = "", string crewName = "", int betAmount = 0, bool? rawGamerName = null)
+        {
+            var winLose = hasWon ? "CELEB_WINNER" : "CELEB_LOSER";
```

Comment:
winLose should be string instead of var. Also this can probably be completely removed as is only needed once.